sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs

```diff
@@ -45,99 +37,41 @@ public async Task VerifyImdsRequestMockAsync()
 
                 AccessToken actualToken = await credential.GetTokenAsync(new TokenRequestContext(MockScopes.Default));
 
-                Assert.AreEqual(expectedToken, actualToken.Token);
+                Assert.AreEqual(ExpectedToken, actualToken.Token);
 
                 MockRequest request = mockTransport.Requests[0];
 
                 string query = request.Uri.Query;
 
                 Assert.IsTrue(query.Contains("api-version=2018-02-01"));
-
                 Assert.IsTrue(query.Contains($"resource={Uri.EscapeDataString(ScopeUtilities.ScopesToResource(MockScopes.Default))}"));
-
                 Assert.IsTrue(request.Headers.TryGetValue("Metadata", out string metadataValue));
-
+                Assert.IsTrue(query.Contains($"client_id=mock-client-id"));
                 Assert.AreEqual("true", metadataValue);
             }
         }
 
         [NonParallelizable]
         [Test]
-        public async Task VerifyImdsRequestWithClientIdMockAsync()
+        [TestCase(400)]
+        [TestCase(502)]
+        public void VerifyImdsRequestHandlesFailedRequestWithCredentialUnavailableExceptionMockAsync(int responseCode)
         {
             using (new TestEnvVar(new () { { "MSI_ENDPOINT", null }, { "MSI_SECRET", null } }))
             {
-                var response = new MockResponse(200);
-
-                var expectedToken = "mock-msi-access-token";
-
-                response.SetContent($"{{ \"access_token\": \"{expectedToken}\", \"expires_on\": \"3600\" }}");
-
+                var response = CreateMockResponse(responseCode, ExpectedToken);
                 var mockTransport = new MockTransport(response);
-
                 var options = new TokenCredentialOptions() { Transport = mockTransport };
-
+                options.Retry.MaxRetries = 0;
```

Comment:
Do we still want to retry on 502 or should it be excluded in the response classifier for the ManagedIdentityCredential? I'm ok with still retrying if we think that's the right thing to do here, but I want to make sure it's an explicit choice.